exampleLitActions/src/conditionalSigning.ts

```diff
@@ -0,0 +1,238 @@
+      console.log(`ℹ️  PKP public key: ${pkpPublicKey}`);
+      console.log(`ℹ️  PKP ETH address: ${pkpInfo.ethAddress}`);
+    } else {
+      console.log(`ℹ️  Using provided PKP: ${pkpPublicKey}`);
+    }
+
+    // Handle Capacity Credit
+    let capacityTokenId = LIT_CAPACITY_CREDIT_TOKEN_ID;
+    if (!capacityTokenId) {
+      console.log("🔄 No Capacity Credit provided, minting a new one...");
+      capacityTokenId = (
+        await litContracts.mintCapacityCreditsNFT({
+          requestsPerKilosecond: 10,
+          daysUntilUTCMidnightExpiration: 1,
+        })
+      ).capacityTokenIdStr;
+      console.log(`✅ Minted new Capacity Credit with ID: ${capacityTokenId}`);
+    } else {
+      console.log(
+        `ℹ️  Using provided Capacity Credit with ID: ${capacityTokenId}`
+      );
+    }
+
+    // Create capacity delegation auth signature
+    console.log("🔄 Creating capacityDelegationAuthSig...");
+    const { capacityDelegationAuthSig } =
+      await litNodeClient.createCapacityDelegationAuthSig({
+        dAppOwnerWallet: ethersWallet,
+        capacityTokenId,
+        delegateeAddresses: [ethersWallet.address],
+        uses: "1",
+      });
+    console.log("✅ Capacity Delegation Auth Sig created");
+
+    // Get session signatures
+    const sessionSigs = await litNodeClient.getSessionSigs({
+      chain: CHAIN_TO_CHECK_CONDITION_ON,
+      capabilityAuthSigs: [capacityDelegationAuthSig],
+      expiration: new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString(), // 24 hours
+      resourceAbilityRequests: [
+        {
+          resource: new LitActionResource("*"),
+          ability: LIT_ABILITY.LitActionExecution,
+        },
+      ],
+      authNeededCallback: async ({ resourceAbilityRequests, expiration, uri }) => {
+        const toSign = await createSiweMessage({
+          uri: uri || "",
+          expiration: expiration || "",
+          resources: resourceAbilityRequests || [],
+          walletAddress: await ethersWallet.getAddress(),
+          nonce: await litNodeClient.getLatestBlockhash(),
+          litNodeClient: litNodeClient,
+        });
+
+        return await generateAuthSig({
+          signer: ethersWallet,
+          toSign,
+        });
+      },
+    });
+
+    // Create auth signature for the action
+    const authSig = await (async () => {
+      const toSign = await createSiweMessage({
+        uri: "http://localhost",
+        expiration: new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString(), // 24 hours
+        walletAddress: await ethersWallet.getAddress(),
+        nonce: await litNodeClient.getLatestBlockhash(),
+        resources: [
+          {
+            resource: new LitActionResource("*"),
+            ability: LIT_ABILITY.LitActionExecution,
+          },
+        ],
+        litNodeClient: litNodeClient,
+      });
+      return await generateAuthSig({
+        signer: ethersWallet,
+        toSign,
+      });
+    })();
```

Comment:
Another util methods that takes the `resources` as an array of Lit Resources (type is importable from our types package or auth package)